decepticonlp/transforms/perturbations.py

```diff
@@ -0,0 +1,317 @@
+            "c": ["x", "d", "f", "v"],
+            "d": ["s", "e", "r", "f", "c", "x"],
+            "e": ["w", "s", "d", "r"],
+            "f": ["d", "r", "t", "g", "v", "c"],
+            "g": ["f", "t", "y", "h", "b", "v"],
+            "h": ["g", "y", "u", "j", "n", "b"],
+            "i": ["u", "j", "k", "o"],
+            "j": ["h", "u", "i", "k", "n", "m"],
+            "k": ["j", "i", "o", "l", "m"],
+            "l": ["k", "o", "p"],
+            "m": ["n", "j", "k", "l"],
+            "n": ["b", "h", "j", "m"],
+            "o": ["i", "k", "l", "p"],
+            "p": ["o", "l"],
+            "q": ["w", "a", "s"],
+            "r": ["e", "d", "f", "t"],
+            "s": ["w", "e", "d", "x", "z", "a"],
+            "t": ["r", "f", "g", "y"],
+            "u": ["y", "h", "j", "i"],
+            "v": ["c", "f", "g", "v", "b"],
+            "w": ["q", "a", "s", "e"],
+            "x": ["z", "s", "d", "c"],
+            "y": ["t", "g", "h", "u"],
+            "z": ["a", "s", "x"],
+        }
+
+        # insert typo
+        for pos in positions_to_shift:
+            # no typo insertion for special characters
+            try:
+                typo_list = keys_in_proximity[word[pos]]
+                word[pos] = random.choice(typo_list)
+            except:
+                break
+
+        # reinsert capitalization
+        for i in range(len(word)):
+            if capitalization[i]:
+                word[i] = word[i].upper()
+
+        # recombine
+        word = "".join(word)
+
+        return word
+
+
+class VisuallySimilarCharacterPerturbations(CharacterPerturbations):
+    """
+        A class used to apply visually similar character perturbations.
+        Methods
+        -------
+        apply(self, word: str, **kwargs)
+            - applies the visually similar perturbation on the word and returns it.
+    """
+
+    def apply(self, word: str, **kwargs):
+        """
+            unicode_array is a list of different unicodes.
+            each char of the word is perturbed by a unicode chosen at random
+            from the unicode_array.
+
+            :word: word to be edited
+            :ignore: default (True), boolean if assertions should be ignored
+
+            eg:
+            input : adversarial
+            output : a̐d̅v̕e̒ŕŝa̅r̕îál̂
+
+            visual_similar_chars("Hey Stop")
+            Hey Stop
+
+            visual_similar_chars("Hey Stop", ignore=False)
+            assertion error
+            """
+        if kwargs.get("ignore", self.get_ignore_default_value()) and " " in word:
+            return word
+        assert " " not in word, self.get_string_not_a_word_error_msg()
+
+        unicode_array = np.array(
+            [u"\u0301", u"\u0310", u"\u0305", u"\u0315", u"\u0312", u"\u0302"]
+        )
+
+        char_array = np.array(list(word))
+
+        int_pick = np.random.randint(0, high=unicode_array.shape[0], size=len(word))
+
+        picked_unicode = unicode_array[int_pick]
```

Comment:
`np.random.choice` can be used here, I think

Reply:
The same goes here. I just refactored the code, this function was already present. I just wrapped it into a class.

Reply:
Gotcha

Reply:
@shashank-m was supposed to be working on this https://github.com/SforAiDl/decepticonlp/issues/28